### PR TITLE
Added SignUpForm

### PR DESCRIPTION
**Pull request by LamboCreeper**
In the process of creating `<SignUpForm />` I have created various other components. This makes the site more modular. This more "rounded" design will be brung into the design of other components too from now.

Here is a screenshot of `<SignUpForm />` in a `<ModalContainer />` on the Resources page:
image

### CODE REVIEW

**Review by Jason2605 — CHANGES_REQUESTED** [review]
From a quick look it seems as though there is no obvious way to close the modal, perhaps a cross, and picking up clicks outside of the modal would be a nice addition to this!

Small thing, when entering what seems to be a valid display name, after clicking out of the input, the border still seems to be red, which to me would indicate a validation error.
    - Seems to happen on passwords too actually!

Another note, i guess more of a question, is it seems to open a modal relating to registration on the "Log in" button. I personally would expect a "log in" and a "sign up" button, which link to two separate modals, what do you think?

**Comment by LamboCreeper** [comment]
> From a quick look it seems as though there is no obvious way to close the modal, perhaps a cross, and picking up clicks outside of the modal would be a nice addition to this!
> 
> Another note, i guess more of a question, is it seems to open a modal relating to registration on the "Log in" button. I personally would expect a "log in" and a "sign up" button, which link to two separate modals, what do you think?

Will add the close button now, good point about separate sign up / login modals - I will do login in a seperate pull request.

File: src/components/organisms/SignUpForm.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    </h1>
+                </header>
+                <LabelWithInput
+                    title="Display Name"
+                    description="This is what you'll be known as on CodeSupport."
+                    input={{
+                        name: "alias",
+                        type: "text",
+                        placeholder: "richardh",
+                        pattern: /^[a-z]{3,10}$/,
```
Comment (Jason2605) [inline src/components/organisms/SignUpForm.js]: This pattern is a bit restrictive, think capital letters.

**Review by Jason2605 — APPROVED** [review]
Nice addition 👍

**Pull request merged**